sig-api-machinery/extensible-admission-to-beta.md

```diff
@@ -0,0 +1,363 @@
+  1. rest.BeforeUpdate
+    1. strategy.PrepareForUpdate
+    1. validation.ValidateObjectMetaUpdate
+    1. strategy.ValidateUpdate
+    1. strategy.Canonicalize
+  1. e.Storage.GuaranteedUpdate
+  1. e.AfterUpdate
+  1. e.Decorator
+
+#### Proposed behavior
+
+1. decode request
+1. admit.ApplyMutations
+1. r.BeforeUpdate
+  1. If strategy-based, call rest.BeforeUpdate
+    1. strategy.PrepareForUpdate
+  1. Otherwise, just do whatever is in the rest.Updater
+1. r.ValidateUpdate
+  1. If strategy-based, call rest.ValidateUpdate
+    1. validation.ValidateObjectMetaUpdate
+    1. strategy.ValidateUpdate
+    1. strategy.Canonicalize
+  1. Otherwise, just do whatever is in the rest.Updater
+1. admit.Validate
+1. r.Update
+  1. For strategy-based handlers:
+    1. e.Storage.GuaranteedUpdate
+    1. e.AfterUpdate
+    1. e.Decorator
+  1. Otherwise, just do whatever is in the rest.Updater
+
+Notes:
+
+- The `rest.Updater` interface's Update method has been split into 3:
+  - BeforeUpdate
+  - ValidateUpdate
+  - Update
+
+
+### Patch
+
+Patch is similar to Update, and similar changes are required to split the call
+to `Update` into `BeforeUpdate`, `ValidateUpdate`, and `Update`. Additionally,
+calls to `admit.Validate` will
+occur in appropriate places within the request lifecycle.
+
+### Delete
+
+#### Current behavior
+
+1. decode request (DeleteOptions)
+2. admit.Admit
+3. r.Delete (the items below are part of the generic registry store / strategy-based storage, `pkg/registry/generic/registry/store.go`)
+  4. e.Storage.Get
+  1. rest.BeforeDelete
+  1. e.Storage.Delete
+  1. e.AfterDelete
+  1. e.Decorator
+
+#### Proposed behavior
+
+1. decode request (DeleteOptions)
+2. r.Get
+1. r.BeforeDelete
+  2. If strategy-based, call rest.BeforeDelete
+  1. Otherwise, just do whatever is in the rest.Deleter
+1. admit.Validate
+1. r.Delete
+  2. For strategy-based handlers:
+    1. e.Storage.Delete
+    1. e.AfterDelete
+    1. e.Decorator
+
+### DeleteCollection
+
+#### Current behavior
+
+1. admit.Admit
+2. decode ListOptions from request parameters
+3. decode request body (DeleteOptions)
+4. r.DeleteCollection (the items below are part of the generic registry store / strategy-based storage, `pkg/registry/generic/registry/store.go`)
+  5. e.List
+  1. run *n* worker goroutines in parallel:
+    1. e.Delete (same as Delete -> Current behavior -> 3)
+
+#### Proposed behavior
+
+1. admit.Validate
+2. decode ListOptions from request parameters
+3. decode request body (DeleteOptions)
+4. r.DeleteCollection ...
+
+### Connect
+
+#### Current behavior
+
+1. connecter.NewConnectOptions
+2. decode request connect options
+3. admit.Admit
+4. connecter.Connect().ServeHTTP()
+
+#### Proposed behavior
+
+1. connecter.NewConnectOptions
+2. decode request connect options
+1. admit.Validate
+2. connecter.Connect().ServeHTTP()
+
+## Ability to run out-of-tree plugins
```

Comment:
Are you talking about a separate plugin mechanism here, or do you mean this will plug in through admission webhooks?

Reply:
uh, old doc.  Let me scrub that section out